### PR TITLE
Attach more time information to helium packets

### PR DESCRIPTION
**Pull request by Vagabond**

### CODE REVIEW

File: src/packet.proto [hunk 1]
```diff
@@ -2,6 +2,23 @@ syntax = "proto3";
...
+    int32 altitude = 3; // meters?
+    uint32 time_accuracy = 4; //nanoseconds
+}
+
+// https://www.eecis.udel.edu/~mills/ntp/html/ntpq.html#system
+message ntp_info {
+    bool synced = 1;
+    uint32 stratum = 2;
+    float sys_jitter = 3; //combined system jitter, in milliseconds
+    float clock_jitter = 4; //clock jitter, in milliseconds
```
Comment (madninja) [inline src/packet.proto]: Do these have to be floats? vs some other fixed point representation?

File: src/packet.proto [hunk 2]
```diff
@@ -2,6 +2,23 @@ syntax = "proto3";
 
 package helium;
 
+message gps_info {
+    uint32 horizontal_accuracy = 1; // meters? this will take into account h3 precision and distance from asserted location
+    uint32 vertical_accuracy = 2; // meters?
+    int32 altitude = 3; // meters?
+    uint32 time_accuracy = 4; //nanoseconds
```
Comment (madninja) [inline src/packet.proto]: Should this be uint64?

File: src/packet.proto [hunk 3]
```diff
@@ -10,9 +27,14 @@ message packet {
     }
     packet_type type = 2;
     bytes payload = 3;
-    uint64 timestamp = 4;
+    uint64 tmst = 4; // internal timer of concentrator, only useful for reply timing
```
Comment (vihu) [inline src/packet.proto]: Can we rename this to `internal_conc_timer`? Or `conc_timer` Or something else perhaps a bit more descriptive?

File: src/packet.proto (hunk 2 above)
Comment (Vagabond) [reply src/packet.proto]: it is typically in the range of 20-200ns but maybe?

File: src/packet.proto (hunk 1 above)
Comment (Vagabond) [reply src/packet.proto]: ntp supplies them as floats

File: src/packet.proto (hunk 2 above)
Comment (JayKickliter) [reply src/packet.proto]: It's sourced from UBX NAV-PVT which represents it as a u32

```tAcc ns Time accuracy estimate (UTC)```

File: src/packet.proto [hunk 4]
```diff
@@ -2,6 +2,23 @@ syntax = "proto3";
...
+message gps_info {
+    float horizontal_accuracy = 1; // meters this will take into account h3 precision and distance from asserted location
+    float vertical_accuracy = 2; // meters
+    int32 altitude = 3; // meters
+    uint64 time_accuracy = 4; //nanoseconds
+}
+
+// https://www.eecis.udel.edu/~mills/ntp/html/ntpq.html#system
+message ntp_info {
+    bool synced = 1;
```
Comment (vihu) [inline src/packet.proto]: So this means that it has synced with some ntp server? Don't see it in the ntp system link above, just wondering.

File: src/packet.proto [hunk 5]
```diff
@@ -2,6 +2,23 @@ syntax = "proto3";
...
+    uint32 stratum = 2;
```
Comment (vihu) [inline src/packet.proto]: Maybe add a comment here that `stratum` is 1-15 as mentioned by ntp standard. Unsure if both are inclusive.

File: src/packet.proto [hunk 6]
```diff
@@ -2,6 +2,23 @@ syntax = "proto3";
...
+    float sys_jitter = 3; // combined system jitter, in milliseconds
+    float clock_jitter = 4; // clock jitter, in milliseconds
```
Comment (vihu) [inline src/packet.proto]: nitpicking, rename to `clk_jitter` to follow the ntp standard

File: src/packet.proto [hunk 7]
```diff
@@ -2,6 +2,23 @@ syntax = "proto3";
...
+    float clock_wander = 5; // clock frequency wander, in PPM
```
Comment (vihu) [inline src/packet.proto]: nitpicking, rename to `clk_wander` same as ntp

**Comment by madninja** [comment]
@Vagabond are we leaving this open until progress is made on other timesources? or can this get closed?

**Pull request closed without merging**